### PR TITLE
feat: weak sup. for weakly supervised dataset

### PR DESCRIPTION
**Pull request by bergr7**
Weak supervision code for weakly supervised ag news datataset.

Note I'm still trying to figure out how to share Rubrix's dataset and rules.

### CODE REVIEW

**Review by alekshiidenhovi — APPROVED** [review]
Fine to merge by me, we can create a separate PR for sharing the dataset and rules, when that is figured out.

**Comment by EdAbati** [comment]
Hey @bergr7 ,
is the error that you mentioned in Discord present in this notebook too?

File: docker-compose.yml [hunk 1]
```diff
@@ -107,11 +109,70 @@ services:
...
     # Allow setting it to false for testing
     external: ${TRAEFIK_PUBLIC_NETWORK_IS_EXTERNAL-true}
+
+volumes:
+  # app-db-data-test:
+  elasticdata:
+
+networks:
+  rubrix:
+    driver: bridge
```
Comment (EdAbati) [inline docker-compose.yml]: @diegoquintanav there seems to be an error here. Please check CI log

**Review by EdAbati — CHANGES_REQUESTED** [review]
Hey @bergr7 all looks good! :)

I fixed a small thing in the Dockerfile (it was not copying requirements before)
Waiting for @diegoquintanav to fix the docker-compose and then we can merge.

Side question, why are we using a validation set with the original labels?
One may have removed the labels from the entire `'train'` set, added labels using weak supervision and only _then_ split between train and validation. In this case, the 'test' set would have been the only one with original labels. Maybe we can include a couple of lines in the notebook/readme explaining this decision?

File: docker-compose.yml (hunk 1 above)
Comment (bergr7) [reply docker-compose.yml]: > Side question, why are we using a validation set with the original labels?
> One may have removed the labels from the entire 'train' set, added labels using weak supervision and only then split between train and validation. In this case, the 'test' set would have been the only one with original labels. Maybe we can include a couple of lines in the notebook/readme explaining this decision?

If I'm not wrong this is explained in the notebook. Of course, you can go ahead and create LFs without a single label. However, you won't be able to measure how good your LFs could be. This means you need significant domain expertise or build it buy expending quite a bit of time on inspection of the data. In a real-world scenario, I'd hand-label a validation set and a test set. I would use the validation set to create the LFs and the test set to test the final model and make sure it generalises. That's why we are using the gold labels in the validation set.

**Comment by bergr7** [comment]
> Hey @bergr7 , is the error that you mentioned in Discord present in this notebook too?

Yes, I need to push the revised version.

File: docker-compose.yml (hunk 1 above)
Comment (EdAbati) [reply docker-compose.yml]: > If I'm not wrong this is explained in the notebook.

Apologies I didn't understand from the notebook that the goal was to measure the quality of the LFs. It makes more sense now! :) Maybe can you add this comment in the notebook in the paragraph `Splits`?

Comment (diegoquintanav) [reply docker-compose.yml]: > 

Why do you think there is an error here?

Comment (diegoquintanav) [reply docker-compose.yml]: Ok, I had two `networks` in the docker-compose file, so I merged them into one.

**Comment by diegoquintanav** [comment]
can we get eyes on this so we can merge this PR?

**Comment by EdAbati** [comment]
@diegoquintanav waiting for @bergr7 to push the fixed Jupiter notebook and the I will review

**Comment by bergr7** [comment]
@EdAbati @diegoquintanav Pushed the fixes. Sorry for the delay.

**Review by EdAbati — APPROVED** [review]
